pkg/svcat/service-catalog/broker.go

```diff
@@ -41,12 +41,21 @@ type Broker interface {
 	// GetURL returns the broker's URL.
 	GetURL() string
 
+	// GetSpec returns the broker's spec.
+	GetSpec() v1beta1.CommonServiceBrokerSpec
+
 	// GetStatus returns the broker's status.
 	GetStatus() v1beta1.CommonServiceBrokerStatus
 }
 
 // Deregister deletes a broker
-func (sdk *SDK) Deregister(brokerName string) error {
+func (sdk *SDK) Deregister(brokerName string, scopeOpts *ScopeOptions) error {
+	if scopeOpts.Scope.Matches(NamespaceScope) {
+		err := sdk.ServiceCatalog().ServiceBrokers(scopeOpts.Namespace).Delete(brokerName, &v1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("deregister request failed (%s)", err)
+		}
+	}
 	err := sdk.ServiceCatalog().ClusterServiceBrokers().Delete(brokerName, &v1.DeleteOptions{})
```

Comment:
This should be wrapped in an if statement, e.g. `scopeOpts.Scope.Matches(ClusterScope)` so that it's not always executed when you the scope is namespace.